src/index.ts

```diff
@@ -63,71 +114,34 @@ export const getTestPostgresDatabaseFactory = <
 
   const workerPromise = getWorker(initialData, options as any)
 
-  const getTestPostgresDatabase: GetTestPostgresDatabase<Params> = async (
-    t: ExecutionContext,
-    params: any,
-    getTestDatabaseOptions?: GetTestPostgresDatabaseOptions
-  ) => {
-    const mapWorkerConnectionDetailsToConnectionDetails = (
-      connectionDetailsFromWorker: ConnectionDetailsFromWorker
-    ): ConnectionDetails => {
-      const pool = new Pool({
-        connectionString: connectionDetailsFromWorker.connectionString,
-      })
-
-      let pgbouncerPool: Pool | undefined
-      if (connectionDetailsFromWorker.pgbouncerConnectionString) {
-        pgbouncerPool = new Pool({
-          connectionString:
-            connectionDetailsFromWorker.pgbouncerConnectionString,
-        })
-      }
-
-      t.teardown(async () => {
-        try {
-          await pool.end()
-          await pgbouncerPool?.end()
-        } catch (error) {
-          if (
-            (error as Error).message.includes(
-              "Called end on pool more than once"
-            )
-          ) {
-            return
-          }
+  const mapWorkerConnectionDetailsToConnectionDetails = (
+    connectionDetailsFromWorker: ConnectionDetailsFromWorker
+  ): ConnectionDetails => {
+    const pool = new Pool({
+      connectionString: connectionDetailsFromWorker.connectionString,
+    })
 
-          throw error
-        }
+    let pgbouncerPool: Pool | undefined
+    if (connectionDetailsFromWorker.pgbouncerConnectionString) {
+      pgbouncerPool = new Pool({
+        connectionString: connectionDetailsFromWorker.pgbouncerConnectionString,
       })
-
-      return {
-        ...connectionDetailsFromWorker,
-        pool,
-        pgbouncerPool,
-      }
     }
 
-    const worker = await workerPromise
-    await worker.available
-
-    const waitForAndHandleReply = async (
-      message: SharedWorker.Plugin.PublishedMessage
-    ): Promise<GetTestPostgresDatabaseResult> => {
-      let reply = await message.replies().next()
-      const replyData: MessageFromWorker = reply.value.data
-
-      if (replyData.type === "RUN_HOOK_BEFORE_TEMPLATE_IS_BAKED") {
-        let result: FinishedRunningBeforeTemplateIsBakedHookMessage["result"] =
-          {
-            status: "success",
-            result: undefined,
-          }
+    return {
+      ...connectionDetailsFromWorker,
+      pool,
+      pgbouncerPool,
+    }
+  }
 
+  let rpcCallback: (data: any) => void
+  const rpc = createBirpc<SharedWorkerFunctions, TestWorkerFunctions>(
```

Comment:
is the idea meant to establish a bidirectional communication with the postgres docker container?

Reply:
not quite

AVA's architecture has one worker thread per test file (called a test worker) and one worker thread per "shared worker" (called... a shared worker)

previous to this PR, we implemented 2-way RPC calls between the test worker and shared workers ourselves, but using birpc really makes the intent and logic cleaner

Reply:
oh i see, this is to communicate with ava's worker thread